src/main/java/me/senseiwells/essentialclient/feature/BetterAccurateBlockPlacement.java

```diff
@@ -98,4 +103,42 @@ public static void sendLookPacket(ClientPlayNetworkHandler networkHandler, Clien
 			playerEntity.isOnGround()
 		));
 	}
+
+	public static Direction[] getFacingOrder() {
+		float f = fakePitch * 0.017453292F;
+		float g = -fakeYaw * 0.017453292F;
+		float yComponent = MathHelper.sin(f);
+		float horizontalComponent = MathHelper.cos(f);
+		float horizontalDecompose1 = MathHelper.sin(g);
+		float horizontalDecompose2 = MathHelper.cos(g);
+		boolean isPositiveX = horizontalDecompose1 > 0.0F;
+		boolean isPositiveY = yComponent < 0.0F;
+		boolean isPositiveZ = horizontalDecompose2 > 0.0F;
+		float absXComponent = isPositiveX ? horizontalDecompose1 : -horizontalDecompose1;
+		float absYCartesian = isPositiveY ? -yComponent : yComponent;
+		float absZComponent = isPositiveZ ? horizontalDecompose2 : -horizontalDecompose2;
+		float absXCartesian = absXComponent * horizontalComponent;
+		float absZCartesian = absZComponent * horizontalComponent;
+		Direction axisVector1 = isPositiveX ? Direction.EAST : Direction.WEST;
+		Direction axisVector2 = isPositiveY ? Direction.UP : Direction.DOWN;
+		Direction axisVector3 = isPositiveZ ? Direction.SOUTH : Direction.NORTH;
+		if (absXComponent > absZComponent) {
+			if (absYCartesian > absXCartesian) {
+				return listClosest(axisVector2, axisVector1, axisVector3);
+			}
+			else {
+				return absZCartesian > absYCartesian ? listClosest(axisVector1, axisVector3, axisVector2) : listClosest(axisVector1, axisVector2, axisVector3);
+			}
+		}
+		else if (absYCartesian > absZCartesian) {
+			return listClosest(axisVector2, axisVector3, axisVector1);
+		}
+		else {
+			return absXCartesian > absYCartesian ? listClosest(axisVector3, axisVector1, axisVector2) : listClosest(axisVector3, axisVector2, axisVector1);
+		}
```

Comment:
This is super messy and hard to read. Have a look at: https://github.com/senseiwells/ArucasLibraries/blob/master/libs/minecraft/PosUtils.arucas#L78-L116. (In arucas but that doesn't really matter.